scripts/identify-web-features.js

```diff
@@ -0,0 +1,303 @@
+  const specUrls = urls
+    .filter(url => {
+      // Filter out known non-spec URLs.
+      return !url.includes("bugzilla") &&
+        !url.includes("github.com") &&
+        !url.includes("webkit.org") &&
+        !url.includes("developer.mozilla.org") &&
+        !url.includes("developer.chrome.com") &&
+        !url.includes("wpt.fyi") &&
+        !url.includes("css-tricks") &&
+        !url.includes("webstatus.dev") &&
+        !url.includes("learn.microsoft.com") &&
+        !url.includes("chromium.org");
+    }).map(url => {
+      // Separate the # from the URL if it exists.
+      return [url, url.split("#")[0]];
+    });
+
+  // Iterate over web features to find spec URL matches.
+  for (const [candidateUrl, candidateUrlNoAnchor] of specUrls) {
+    for (const id in features) {
+      const feature = features[id];
+      const featureSpecs = Array.isArray(feature.spec) ? feature.spec : [feature.spec];
+      if (featureSpecs.some(spec => spec === candidateUrl || spec === candidateUrlNoAnchor)) {
+        gatheredFeatures.add(id)
+      }
+    }
+  }
+}
+
+// Identify web-features based on explorer URLs in the issue body.
+function gatherFeaturesFromExplorerUrls(issueBody, gatheredFeatures) {
+  // https://web-platform-dx.github.io/web-features-explorer/features/<feature-id>
+  let urls = issueBody.match(/https?:\/\/web-platform-dx.github.io\/web-features-explorer\/features\/[a-z0-9-]+/g) || [];
+  urls = urls.map(url => url.split("#")[0]);
+
+  for (const url of urls) {
+    // Extract the feature ID from the URL.
+    const match = url.match(/features\/([a-z0-9-]+)/);
+    if (match && match[1]) {
+      const id = match[1];
+      const feature = features[id];
+      if (feature) {
+        gatheredFeatures.add(id);
+      }
+    }
+  }
+}
+
+// Identify web-features based on WPT URLs in the issue body.
+function gatherFeaturesFromWPTUrls(issueBody, gatheredFeatures) {
+  // https://wpt.fyi/results/?q=feature:<feature-id>
+  let urls = issueBody.match(/https?:\/\/wpt\.fyi\/results\/\?q=feature:[a-z0-9-]+/g) || [];
+  urls = urls.map(url => url.split("#")[0]);
+
+  for (const url of urls) {
+    // Extract the feature ID from the URL.
+    const match = url.match(/feature:([a-z0-9-]+)/);
+    if (match && match[1]) {
+      const id = match[1];
+      const feature = features[id];
+      if (feature) {
+        gatheredFeatures.add(id);
+      }
+    }
+  }
+}
+
+// Given a GitHub issue, find the web-features that are referenced in the issue body.
+// For now, we only look for URLs that match the web-features specifications.
+function findFeaturesInIssue(issue) {
+  const features = new Set();
+  
+  gatherFeaturesFromSpecUrls(issue.body, features);
+  gatherFeaturesFromExplorerUrls(issue.body, features);
+  gatherFeaturesFromWPTUrls(issue.body, features);
+  // TODO: Match on other things too:
+  // - web-features ID in the body of the issue, which might be there since the issue template asks for it.
```

Comment:
I think this one is important to support and should _override_ features we get from other sources. That means that the author (or whoever can edit the topmost comment) will be in full control.